src/Hint/NewType.hs

```diff
@@ -1,19 +1,24 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE PackageImports #-}
 {-
     Suggest newtype instead of data for type declarations that have
     only one field. Don't suggest newtype for existentially
     quantified data types because it is not valid.
 
 <TEST>
 data Foo = Foo Int -- newtype Foo = Foo Int
+data Foo where Foo :: Int -> Foo -- newtype Foo where Foo :: Int -> Foo
 data Foo = Foo Int deriving (Show, Eq) -- newtype Foo = Foo Int deriving (Show, Eq)
 data Foo = Foo { field :: Int } deriving Show -- newtype Foo = Foo { field :: Int } deriving Show
 data Foo a b = Foo a -- newtype Foo a b = Foo a
 data Foo = Foo { field1, field2 :: Int}
 data S a = forall b . Show b => S b
 data Color a = Red a | Green a | Blue a
 data Pair a b = Pair a b
+data Pair a b where Pair :: a -> b -> Pair a b
 data Foo = Bar
 data Foo a = Eq a => MkFoo a
+data Foo a = () => Foo a -- newtype Foo a = Foo a
```

Comment:
This case isn't currently covered.

This has very weird behaviour, unfortunately. It counts as `Just (L _ [])` for a context, but outputting it results in what would be `Nothing` if it were to be parsed again (`show . read` isn't `id` in other words). As a result the suggestion comes out without the `() =>` part.

It seems to be mostly a positive change (I don't know of use cases for `() =>` contexts), but I still have to ask, is this fine @ndmitchell?

Reply:
Yep, absolutely 100% fine. Agreed, it's an improvement.